### PR TITLE
Make menu work in WebComponent (continued)

### PR DESCRIPTION
**Pull request by emielbeinema**
### Link to related issue (if applicable)
#1415 

### Summary of proposed changes
#1416 introduced some browser-compatibility issues because of the use of non-standard extensions. This pull request aims to fix that (tested on recent Chrome, Firefox, Edge)

### Checklist
- [x] Use `develop` as the base branch
- [x] Exclude the gulp build (`/dist` changes) from the PR
- [x] Test on [supported browsers](https://github.com/sampotts/plyr#browser-support)

### CODE REVIEW

File: src/js/controls.js [hunk 1]
```diff
@@ -1139,8 +1139,9 @@ const controls = {
             show = false;
         } else if (is.event(input)) {
             // If Plyr is in a shadowDOM, the event target is set to the component, instead of the
-            // element in the shadowDOM. The path, however, is complete.
-            const isMenuItem = popup.contains(input.path[0]);
+            // element in the shadowDOM. The path, if available, is complete.
+            const target = typeof input.composedPath === 'function' ? input.composedPath()[0] : input.target;
```
Comment (sampotts) [inline src/js/controls.js]: You can use `is.function(input.composedPath)` here

Comment (emielbeinema) [reply src/js/controls.js]: Of course, thanks!

File: src/js/utils/elements.js [hunk 2]
```diff
@@ -234,10 +234,10 @@ export function matches(element, selector) {
...
 
     const matches =
-        prototype.matches ||
-        prototype.webkitMatchesSelector ||
-        prototype.mozMatchesSelector ||
-        prototype.msMatchesSelector ||
+        element.matches ||
+        element.webkitMatchesSelector ||
+        element.mozMatchesSelector ||
+        element.msMatchesSelector ||
```
Comment (sampotts) [inline src/js/utils/elements.js]: What was this change for?

**Comment by sampotts** [comment]
I've manually merged the first change but I think the second is a mistake.

**Comment by emielbeinema** [comment]
Thanks for merging this request. As for the second change, however, I do still think this change is necessary (and correct!) and I suspect that selecting a menu option in the shadowDOM will now omit deselecting the previously selected menu option.

Some supportive evidence that the current code doesn't quite work is how it is compiled. In [`dist/plyr.js`](https://github.com/sampotts/plyr/blob/master/dist/plyr.js#L118), we simply see:
`var matches = match;`, always selecting the custom implementation (that poses problems for shadowDOM).

If you have another proposed solution, I would be happy to implement that, but I do hope we can get this fixed one way or another.

**Pull request closed without merging**